File: views/controllers/readController.js
import ReadModel from '../models/readModel.js';
import ReadView from '../views/readView.js';

export default class ReadController{
    constructor(){
        this.model = new ReadModel();
        this.view = new ReadView();
    }

    getNames(){
        this.model.sendNamesRequest(this.updateOptions);
    }

    getTable(name){
        this.model.sendTableRequest(name, this.updateTable);
    }

    updateOptions(data){
        let names = ReadModel.parseNames(data);
        ReadView.updateOptions(names.sort());
    }

    updateTable(data){
        console.log(data);
        ReadView.updateTable(data);
    }

}

File: controllers/readController.py
from models.readModel import ReadModel

class ReadController:
    def __init__(self):
        self.model = ReadModel()

    def names(self):
        print('names')
        names = self.model.getTablesNames()
        return {"names": names}

    def table(self, name):
        print('table')
        table = self.model.getTable(name)
        print(table)
        return {"table": table}
